### PR TITLE
Add Events

### PR DESCRIPTION
**Pull request by DammyShittu**
## In this PR, I;
- created a `logic` directory and added two files (`calculate.js` and `operate.js`),
- imported the files in `Calculator` component,
- implemented event handlers to use `calculate.js` and `operate.js` in `Calculator` component,
- tested the app in the browser to be sure it performed Math operations, and
- fixed all linter errors.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        total: null,
+      };
+    }
+    return {
+      next: buttonName,
+      total: null,
+    };
+  }
+
+  if (buttonName === '.') {
```
Comment (Dipeshtwis) [inline src/logic/calculate.js]: - [ ]  Please work on the division of a number with zero let us say `9/0` normally you have to get error in the screen but currently you are getting this

Screenshot from 2021-09-15 21-00-38

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  if (!obj.next) {
+    return { operation: buttonName };
+  }
+
+  // save the operation and shift 'next' into 'total'
+  return {
+    total: obj.next,
+    next: null,
+    operation: buttonName,
+  };
```
Comment (Dipeshtwis) [inline src/logic/calculate.js]: [OPTIONAL]

- When I tried to do something like `8 +` and then I pressed any operation again like `-`, it's better if it's changed into `8 -` removing the previous operation. But in your case, it's saying invalid which shows in the below screenshot. It's better if you work on that after that your app will be fully bug-free.

Screenshot from 2021-09-15 21-00-53

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @DammyShittu, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: Good use of state and props.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**_Check the comments under the review_**

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/calculate.js (hunk 2 above)
Comment (DammyShittu) [reply src/logic/calculate.js]: Hi @Dipeshtwis ,
Thank you for the review.
I have worked on the requested changes and will proceed to request a review now.

Cheers 👋

**Review by AshakaE — APPROVED** [review]
### Hi @DammyShittu 🙋‍♀️, 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by DammyShittu** [comment]
Thank you @AshakaE 🚀

**Pull request merged**